### ISSUE TITLE
Find a solution to declare peer dependencies

### ISSUE THREAD

**Issue by dylang** [depcheck/depcheck#130]
`npm@2` is installing the peer dependencies which is an unexpected behavior, as they are optional for `depcheck`, and this means when using `npm@2`, they are always installed.

`npm@3` won't install them, but gives unexpected warnings:

```
├── UNMET PEER DEPENDENCY node-sass@^3.4.0
└── UNMET PEER DEPENDENCY typescript@^1.8.0

npm WARN depcheck@0.6.2 requires a peer of node-sass@^3.4.0 but none was installed.
npm WARN depcheck@0.6.2 requires a peer of typescript@^1.8.0 but none was installed.
```

I feel like the feature of using the dependencies when they are available is great, but instead of including them as peer dependencies, just include instructions in the docs what users need to do.

**Comment by lijunle** [comment]
I follow the pattern from [eslint-config-airbnb](https://github.com/airbnb/javascript/tree/master/packages/eslint-config-airbnb). They mark react as peer dependency. There is [some discussions](https://github.com/airbnb/javascript/issues/751) to remove the peer dependency or split the package into three sub-packages. 

In my option, those UMET warnings are fine because they are warnings not errors. I could like to keep this open and wait for a better idea/solution.

**Comment by lijunle** [comment]
Because the peer dependencies is blocking `npm shrinkwrap` (https://github.com/dylang/npm-check/issues/108) caused by NPM bug (https://github.com/npm/npm/issues/10836), I decide to remove the peer dependencies from `package.json` and only keep them in README.

However, let us keep this open and find a better solution to resolve these *optional peer dependencies*.

**Comment by CatzillaOrz** [comment]
try to use ``` nvm ls-remote ``` find a stable node version then use ``` nvm use ```

**Comment by rumpl** [comment]
Is this still an issue?

I'm not even sure I understand what's going on to be honest.

**Comment by nnfans** [comment]
`node-sass` is deprecated and it makes the npm install throw an error
can we remove it or update to `7.0.1` so it does't throw an error
```
npm ERR! code 1
npm ERR! path /Users/ajaib/repo/depcheck/node_modules/node-sass
npm ERR! command failed
npm ERR! command sh -c node scripts/build.js
npm ERR! Building: /Users/ajaib/.nvm/versions/node/v16.14.2/bin/node /Users/ajaib/repo/depcheck/node_modules/node-gyp/bin/node-gyp.js rebuild --verbose --libsass_ext= --libsass_cflags= --libsass_ldflags= --libsass_library=
npm ERR!   c++ -o Release/obj.target/libsass/src/libsass/src/ast.o ../src/libsass/src/ast.cpp '-DNODE_GYP_MODULE_NAME=libsass' '-DUSING_UV_SHARED=1' '-DUSING_V8_SHARED=1' '-DV8_DEPRECATION_WARNINGS=1' '-DV8_DEPRECATION_WARNINGS' '-DV8_IMMINENT_DEPRECATION_WARNINGS' '-D_GLIBCXX_USE_CXX11_ABI=1' '-D_DARWIN_USE_64_BIT_INODE=1' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DOPENSSL_NO_PINSHARED' '-DOPENSSL_THREADS' '-DLIBSASS_VERSION="3.5.5"' -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/include/node -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/src -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/deps/openssl/config -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/deps/openssl/openssl/include -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/deps/uv/include -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/deps/zlib -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/deps/v8/include -I../src/libsass/include  -O3 -gdwarf-2 -mmacosx-version-min=10.7 -arch arm64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=c++11 -stdlib=libc++ -fno-strict-aliasing -MMD -MF ./Release/.deps/Release/obj.target/libsass/src/libsass/src/ast.o.d.raw   -c
npm ERR!   c++ -o Release/obj.target/libsass/src/libsass/src/ast_fwd_decl.o ../src/libsass/src/ast_fwd_decl.cpp '-DNODE_GYP_MODULE_NAME=libsass' '-DUSING_UV_SHARED=1' '-DUSING_V8_SHARED=1' '-DV8_DEPRECATION_WARNINGS=1' '-DV8_DEPRECATION_WARNINGS' '-DV8_IMMINENT_DEPRECATION_WARNINGS' '-D_GLIBCXX_USE_CXX11_ABI=1' '-D_DARWIN_USE_64_BIT_INODE=1' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DOPENSSL_NO_PINSHARED' '-DOPENSSL_THREADS' '-DLIBSASS_VERSION="3.5.5"' -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/include/node -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/src -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/deps/openssl/config -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/deps/openssl/openssl/include -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/deps/uv/include -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/deps/zlib -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/deps/v8/include -I../src/libsass/include  -O3 -gdwarf-2 -mmacosx-version-min=10.7 -arch arm64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=c++11 -stdlib=libc++ -fno-strict-aliasing -MMD -MF ./Release/.deps/Release/obj.target/libsass/src/libsass/src/ast_fwd_decl.o.d.raw   -c
npm ERR!   c++ -o Release/obj.target/libsass/src/libsass/src/backtrace.o ../src/libsass/src/backtrace.cpp '-DNODE_GYP_MODULE_NAME=libsass' '-DUSING_UV_SHARED=1' '-DUSING_V8_SHARED=1' '-DV8_DEPRECATION_WARNINGS=1' '-DV8_DEPRECATION_WARNINGS' '-DV8_IMMINENT_DEPRECATION_WARNINGS' '-D_GLIBCXX_USE_CXX11_ABI=1' '-D_DARWIN_USE_64_BIT_INODE=1' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DOPENSSL_NO_PINSHARED' '-DOPENSSL_THREADS' '-DLIBSASS_VERSION="3.5.5"' -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/include/node -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/src -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/deps/openssl/config -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/deps/openssl/openssl/include -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/deps/uv/include -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/deps/zlib -I/Users/ajaib/Library/Caches/node-gyp/16.14.2/deps/v8/include -I../src/libsass/include  -O3 -gdwarf-2 -mmacosx-version-min=10.7 -arch arm64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=c++11 -stdlib=libc++ -fno-strict-aliasing -MMD -MF ./Release/.deps/Release/obj.target/libsass/src/libsass/src/backtrace.o.d.raw   -c
```
